src/IR.h

```diff
@@ -886,16 +886,44 @@ struct Atomic : public StmtNode<Atomic> {
     static const IRNodeType _node_type = IRNodeType::Atomic;
 };
 
-/** Represent a target-specific vector instruction.
- *  Intrinsic may not be element-wise operation, i.e.
- *  dot_products. */
-struct VectorIntrinsic : public ExprNode<VectorIntrinsic> {
-    std::string name;
+/** Represent a length-agnostic and target-specific
```

Comment:
Also, for the ones targeting machine instructions, is it expected the argument types exactly match the hardware specification? I expect the plan is to support some instructions that can be used on any of a number of types with a single name here. Also the order of the arguments may be difficult to infer from just the name even if one has the hardware spec handy. This will need to be documented somewhere and likely a comment on each name (enum member) is the way to do it.

Reply:
The argument types are expected to be multiples of the hardware specification, so they can be split up into series of the target instructions. 
We already do support instructions with many different possible input types (i.e. `dot_product`), and we expect the backend to have a matching implementation.

> Also the order of the arguments may be difficult to infer from just the name even if one has the hardware spec handy.

The order of the arguments is expected to match the order of the arguments in the intrinsic definition, but I see your point. I'm not sure I have a good solution for this though.

> This will need to be documented somewhere and likely a comment on each name (enum member) is the way to do it.

Is that better than documenting in the intrinsics tables for each backend?